app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedPresenter.java

```diff
@@ -0,0 +1,165 @@
+package ru.vssemikoz.newsfeed.newsfeed;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.util.List;
+
+import retrofit2.Response;
+import ru.vssemikoz.newsfeed.MainApplication;
+import ru.vssemikoz.newsfeed.models.Category;
+import ru.vssemikoz.newsfeed.models.NewsApiResponse;
+import ru.vssemikoz.newsfeed.models.NewsItem;
+import ru.vssemikoz.newsfeed.navigator.Navigator;
+import ru.vssemikoz.newsfeed.data.NewsApiRepository;
+import ru.vssemikoz.newsfeed.data.NewsStorage;
+import ru.vssemikoz.newsfeed.data.mappers.NewsItemMapper;
+
+import static androidx.core.util.Preconditions.checkNotNull;
+
+public class NewsFeedPresenter implements NewsFeedContract.Presenter {
+    private static final String TAG = NewsFeedPresenter.class.getName();
+    private final NewsFeedContract.View view;
+
+    private boolean showOnlyFavorite = false;
+    private Category category = Category.ALL;
+    private List<NewsItem> news;
+
+    private MainApplication mainApplication;
+    private NewsStorage newsStorage;
+    private NewsApiRepository repository;
+
+    public NewsFeedPresenter(NewsFeedContract.View view) {
+        this.view = checkNotNull(view, "tasksView cannot be null!");
+        this.view.setPresenter(this);
+        this.mainApplication = MainApplication.getInstance();
+    }
+
+    @Override
+    public void start() {
+        initStartValues();
+        initApiStorage();
+        initNewsStorage();
+        loadNewsFromApi();
+    }
+
+    private void initApiStorage() {
+        repository = new NewsApiRepository(mainApplication);
+    }
+
+    private void loadNewsFromApi() {
+        view.showProgressBar();
+        processNewsApiResponse();
```

Comment:
`processNewApiResponse` means that you already have a response to be processed. But what you're doing here actually is requesting news so we can rename this method to something like `requestNewsFromApi`.